tox.ini

```diff
@@ -156,11 +156,13 @@ envlist =
     docker-tests
     docs
 
+    generate
+
 [testenv]
 deps =
   -c dev-requirements.txt
-  test: pytest
-  test: pytest-benchmark
+  pytest
+  pytest-benchmark
```

Comment:
With fresh tox envs, running `tox -e propagator` or `tox -e sdkextension` results in tox running the following command:

```
sdkextension installdeps: -cdev-requirements.txt
```
which fails as pip doesn't receive any real requirements to install. 

Running `tox -e test-propagator` fixes the issue as tox matches the `test` dep with the env. This means we'd always have to run such envs with the test- prefix which can be annoying and confuse contributors.

The above change installs pytest and pytest-benchmark by default  in all envs. All envs need pytest anyway so there is no harm in doing this. Plus this always adds at least one dependency to install which works with `pip -c dev-requirements.txt`.

Happy to hear other ideas to address this issue.